### PR TITLE
Add proselint

### PR DESCRIPTION
**Pull request by CJ-Wright**
Proselint

### CODE REVIEW

**Comment by CJ-Wright** [comment]
@conda-forge/staged-recipes I think this might be good to go!

File: recipes/proselint/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+build:
+  number: 0
+  entry_points:
+    - proselint = proselint.command_line:proselint
+  script: "{{ PYTHON }} -m pip install . --no-deps --ignore-installed -vv "
+
+requirements:
+  host:
+    - click
```
Comment (CJ-Wright) [inline recipes/proselint/meta.yaml]: These are listed in "install_requires" so they seem like real host deps

Comment (isuruf) [reply recipes/proselint/meta.yaml]: you only need `setup_requires`

**Comment by CJ-Wright** [comment]
@conda-forge/staged-recipes ready for another round.

File: recipes/proselint/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+package:
+  name: "{{ name|lower }}"
+  version: "{{ version }}"
+
+source:
+  url: "https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz"
+  sha256: "3a87eb393056d1bc77d898e4bcf8998f50e9ad84f7b9ff7cf2720509ac8ef904"
+
+build:
```
Comment (isuruf) [inline recipes/proselint/meta.yaml]: `noarch: python` ?

**Comment by CJ-Wright** [comment]
@conda-forge/staged-recipes ready for another round.

**Comment by CJ-Wright** [comment]
Thank you very much!

**Pull request merged**